IdentityCore/src/util/NSDictionary+MSIDExtensions.m

```diff
@@ -53,16 +53,17 @@ + (NSDictionary *)msidDictionaryFromURLEncodedString:(NSString *)string
     for (NSString *query in queries)
     {
         NSArray *queryElements = [query componentsSeparatedByString:@"="];
-        if (queryElements.count > 2)
+        
+        NSString *key = isFormEncoded ? [queryElements[0] msidTrimmedString].msidWWWFormURLDecode : [queryElements[0] msidTrimmedString].msidURLDecode;
```

Comment:
We should probably make sure that queryElements count is more than 1 before executing this line

Reply:
+1

Reply:
I thought about that, and saw that `componentsSeparatedByString:` will always return at least one element (in this case when there is no '=' in the NSString): https://developer.apple.com/documentation/foundation/nsstring/1413214-componentsseparatedbystring

Reply:
Thanks @juan-arias. Would this be true even if the string is empty?

Reply:
Yes, and that is also covered in the `testmsidDictionaryFromURLEncodedString_whenMalformedQuery_shouldReturnDictWithoutBadQuery` test, when the querystring ends with `&`. It will return @[@""] which is then handled by the "empty key" case and not included in the returning dictionary.

Reply:
Thanks a lot for confirming, @juan-arias!